packages/daimo-common/src/op.ts

```diff
@@ -112,6 +118,37 @@ export interface PaymentLinkOpEvent extends OpEventBase {
   memo?: string;
 }
 
+/**
+ * Represents a token swap between two accounts on the same chain.
+ * Same chain, different coins.
+ *
+ * A token swap can be inbound swap (e.g. a Daimo account receives a foreign
+ * token transfer in their inbox) or outbound swap (e.g. account Alice sends a
+ * foreign token transfer to Bob).
+ */
+export interface SwapClog extends OpEventBase {
+  type: "swap";
+
+  from: Address;
+  to: Address;
+
+  /** TODO: use bigint? Unnecessary for USDC. MAX_SAFE_INT = $9,007,199,254 */
+  amount: number; // amount that affects the user
+
+  /** "Other" coin involved in the swap (i.e. not homeCoin or bridgeCoin) */
```

Comment:
maybe just "not homeCoin", SwapClog doesn't support cross chain yet

Reply:
add some indication of whether we're swapping *to* or *from* the other coin. maybe `amount` already covers this... is `amount` negative for outbound swaps?

Reply:
the way that it was done for OpEventDisplays was checking whether the account was the "from" address or "to" address